rusk-wallet/src/lib/prompt.rs

```diff
@@ -126,121 +136,197 @@ pub(crate) fn welcome() -> u8 {
 }
 
 /// Request the user to select a wallet to open
-pub(crate) fn select_wallet(dir: &str, wallets: Vec<String>) -> PathBuf {
+pub(crate) fn select_wallet(dir: &str, wallets: Vec<String>) -> Option<PathBuf> {
     // select the wallet
     let q = Question::select("wallet")
         .message("Please select the wallet you wish to use:")
         .choices(&wallets)
+        .default_separator()
+        .choice("Other...")
         .build();
     let a = requestty::prompt_one(q).unwrap();
     let wi = a.as_list_item().unwrap().index;
 
-    // gen full path for selected wallet
-    let mut path = PathBuf::new();
-    path.push(dir);
-    path.push(wallets[wi].clone());
-    path
+    if wi > wallets.len() {
+        None
+    } else {
+        // gen full path for selected wallet
+        let mut path = PathBuf::new();
+        path.push(dir);
+        path.push(wallets[wi].clone());
+        Some(path)
+    }
+
+}
+
+/// Request a name for the wallet
+pub(crate) fn request_wallet_name() -> String {
+
+    let q = Question::input("name")
+        .message("Please enter a wallet name:")
+        .validate_on_key(|name, _| !wallet_exists(name))
+        .validate(|name, _| {
+            if !wallet_exists(name) {
+                Ok(())
+            } else {
+                Err("A wallet with this name already exists".to_string())
+            }
+        })
+        .build();
+
+    let a = requestty::prompt_one(q).unwrap();
+    a.as_string().unwrap().to_string()
+}
+
+/// Checks if a wallet with this name already exists
+fn wallet_exists(name: &str) -> bool {
+    let mut pb = PathBuf::new();
+    pb.push(WalletCfg::default_data_dir());
+    pb.push(name);
+    pb.set_extension("dat");
+    pb.is_file()
 }
 
 /// Let the user choose a command to execute
-pub(crate) fn command() -> Option<CliCommand> {
-    let msg = "What would you like to do?";
+pub(crate) fn command(offline: bool) -> Option<CliCommand> {
+
+    // notify the user if we're note connected
+    let offline_notice = match offline {
+        false => "",
+        true => " [offline]",
+    };
+
+    // choices differ if user is online or not
+    let choices = match offline {
```

Comment:
If an "offline" command is always available in the "online" commands, repeating it could be a bitter counterintuitive.
Maybe add set of default commands (the offline) and add the "online" commands on top of them

Reply:
It's just one command that repeats, not worth to "recycle" code imho

Reply:
nvm I found a cleaner way, thanks for the comment!

Reply:
We have "export consensus keys" too

Reply:
Yeah but I'm keeping that option on another category for now :)